Packages/KeyAppKit/Sources/Onboarding/Service/DeviceShare/DeviceShareMigrationService.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  File.swift
+//
+//
+//  Created by Giang Long Tran on 13/06/2023.
+//
+
+import Combine
+import Foundation
+import KeyAppKitCore
+
+public class DeviceShareMigrationService {
+    let errorObserver: ErrorObserver
+    let isMigrationAvailableSubject: CurrentValueSubject<Bool, Never> = .init(false)
+    var subscriptions: [AnyCancellable] = []
+
+    public init(
+        isWeb3AuthUser: AnyPublisher<Bool?, Never>,
```

Comment:
Opinion. I don't like the idea of  passing publishers to the init. It might generate error or completed and it will stop working as well as it can generate a set of events